[KOGITO-4958] Setup Spring Boot/custom probes

Kevin-Mok opened the pull request:
See: https://issues.redhat.com/browse/KOGITO-4958

## Description
With [this PR](https://github.com/kiegroup/kogito-cloud-operator/pull/841), the operator now expects that Spring Boot applications have the [Spring Boot Actuator dependency](https://docs.spring.io/spring-boot/docs/2.3.0.RELEASE/reference/html/production-ready-features.html) imported. This is now clarified in the docs when setting up an example project and in the OpenShift section.

## Changes
- require Spring Boot Actuator dependency (when deploying on OpenShift/Kubernetes) in following sections:
  - creating Kogito Maven project
  - setting up probes
  - deploying on OpenShift web console
  - deploying with Kogito CLI
- remove note about startup probes not being supported (now implemented with [KOGITO-4322](https://github.com/kiegroup/kogito-operator/pull/826))
  - update probe mentions to include startup
- show how to setup custom probe with TCP socket (not used by default in operator any more)
- remove requiring health extension in travel agency tutorial (added to example's `pom.xml` already)

## Related PR's
- [kogito-operator](https://github.com/kiegroup/kogito-cloud-operator/pull/841): [KOGITO-4898] Set probe defaults based on YAML action values
- [kogito-runtimes](https://github.com/kiegroup/kogito-runtimes/pull/1232): [KOGITO-4958] Add Spring Boot Actuator as default dependency
- [kogito-examples](https://github.com/kiegroup/kogito-examples/pull/660): [KOGITO-4958] Add Spring Boot Actuator dependency

sutaakar commented on doc-content/kogito-docs/src/main/asciidoc/creating-running/chap-kogito-creating-running.adoc:
```diff
@@ -583,6 +583,27 @@ $ mvn archetype:generate \
+.SmallRye Health extension for Quarkus applications on {OPENSHIFT}
+[source]
+----
+$ mvn quarkus:add-extension -Dextensions="smallrye-health"
```
This extension is already added in case user creates a project using archetype command from above.
Would it have more sense just to mention to execute this when user doesn't use archetype generation?

sutaakar commented on doc-content/kogito-docs/src/main/asciidoc/creating-running/chap-kogito-creating-running.adoc:
```diff
@@ -594,26 +615,20 @@ $ mvn archetype:generate \
-----
-
-This command generates the following dependency in the `pom.xml` file of your {PRODUCT} project on Quarkus:
+If you plan to run your Spring Boot application on {OPENSHIFT} or Kubernetes, you must also import the https://docs.spring.io/spring-boot/docs/2.3.0.RELEASE/reference/html/production-ready-features.html[Spring Boot Actuator dependency] for the https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes[liveness, readiness, and startup probes], as shown in  ...
```
Same as above

hmanwani-rh reviewed (commented):
@Kevin-Mok I've added a few tweaks to the content. Please take a look and let me know if the changes look good. 
Also, I think we also need a release note for this change, don't we?

Following is the doc preview of Kogito documentation
[Kogito documentation](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-KOGITO-4958/#proc-kogito-enabling-probes_kogito-deploying-on-openshift)

Kevin-Mok commented:
@hmanwani-rh 

> I've added a few tweaks to the content. Please take a look and let me know if the changes look good.

The tweaks look good. I just removed the unnecessary sections @sutaakar mentioned in my latest commit. 

> Also, I think we also need a release note for this change, don't we?

Do you mean the [1.6.0 release notes](https://issues.redhat.com/secure/ReleaseNote.jspa?projectId=12322421&version=12353771)?  Won't [this Jira ticket](https://issues.redhat.com/browse/KOGITO-4958) be automatically added once it's resolved (after the 2 other relevant open PR's are merged)?

hmanwani-rh commented:
> @hmanwani-rh
> 
> > I've added a few tweaks to the content. Please take a look and let me know if the changes look good.
> 
> The tweaks look good. I just removed the unnecessary sections @sutaakar mentioned in my latest commit.
> 
> > Also, I think we also need a release note for this change, don't we?
> 
> Do you mean the [1.6.0 release notes](https://issues.redhat.com/secure/ReleaseNote.jspa?projectId=12322421&version=12353771)? Won't [this Jira ticket](https://issues.redhat.com/browse/KOGITO-4958) be automatically added once it's resolved (after the 2 other relevant open PR's are merged)?

I mean release notes in the Kogito documentation, see this section: https://docs.jboss.org/kogito/release/latest/html_single/#ref-kogito-rn-new-features_kogito-release-notes

I believe we need an entry in the section shared above, wdyt?

hmanwani-rh reviewed (commented):
@Kevin-Mok Added a few tweaks to the release note. Please take a look and share your feedback. 

[Release note for KOGITO-4958](http://file.pnq.redhat.com/~hmanwani/BXMSDOC-KOGITO-4958/#_kogito_operator_and_cli)

Kevin-Mok commented:
@hmanwani-rh I think it looks fine. The only feedback I have is that "Deploy the Spring Boot actuator dependency" sounds strange as we don't deploy the dependency but rather the Kogito application.

hmanwani-rh reviewed (commented):
@Kevin-Mok ah, thanks for clarifying that. How about we change the title as follows.

hmanwani-rh commented:
@Kevin-Mok are we good to merge this PR?

Kevin-Mok commented:
@hmanwani-rh Yes.
